[DOC] Updates to external listener config procedures (#3353)
strimzi/strimzi-kafka-operator: documentation/modules/proc-accessing-kafka-using-loadbalancers.adoc

[id='proc-accessing-kafka-using-loadbalancers-{context}']
= Accessing Kafka using loadbalancers

This procedure describes how to access a Strimzi Kafka cluster from an external client using loadbalancers.

> To connect to a broker, you need a hostname (advertised address) for the loadbalancer _bootstrap address_,

Reviewer: ```suggestion
To connect to a broker, you need the address of the _bootstrap loadbalancer_,
```

> as well as the certificate used for authentication.

Reviewer: ```suggestion
as well as the certificate used for TLS encryption.
```

. Create or update the resource.
+
[source,shell,subs=+quotes]
kubectl apply -f _KAFKA-CONFIG-FILE_
+
`loadbalancer` type services and loadbalancers are created for each Kafka broker, as well as an external _bootstrap service_.

> The bootstrap service routes external traffic to the Kafka brokers.

Reviewer: ```suggestion
The bootstrap service routes external traffic to all Kafka brokers.
```

DNS names and IP addresses used for connection are propagated to the `status` of each service.

> . Retrieve the address of the bootstrap service to access the Kafka cluster.

Reviewer: This whole section / bullet point should be replaced with something like this

```suggestion
. Retrieve the address of the bootstrap service to access the Kafka cluster.

[source,shell,subs=+quotes]
kubectl get kafka _KAFKA-CLUSTER-NAME_ -o=jsonpath='{.status.listeners[?(@.type=="external")].bootstrapServers}{"\n"}'
```

Author: @scholzj I originally changed this and then changed back, thinking the _status_ was the key. I've replaced now, but we don't want both?

Reviewer: What we now basically do is we do the original steps in the operator and put it always into a single place. So instead of taking it from the status of the operator, the user can take it from the Kafka status with all the methods. So the user does not have to search for the services etc.

Author: Thanks. I get it now.
